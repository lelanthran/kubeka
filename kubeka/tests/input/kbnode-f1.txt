[cron]
appname = $(env_read APPNAME)
git_origin = ssh://some-origin/some-repo.git
git_branch = master


[job]
ID = Full deployment
MESSAGE = Performing a full deployment
JOBS[] = [ build, upgrade, Done ]
for_filter = first node

[cron]
ID = build
MESSAGE = Starting build process
JOBS[] = [ Check changes, checkout, Build thing, test, package ]
for_filter = second node

[job]
ID = upgrade
MESSAGE = Performing upgrade
JOBS[] = [ pre-deployment, Upgrade DB, Copy files]
for_filter = third node

[cron]
ID = Check changes
MESSAGE = Checking if sources changed, will continue only if sources changed
EXEC += if [ `git  ls-remote $(git_origin) heads/master | cut -f 1 ` == $(current HASH)]; then
EXEC +=    exit -1;
EXEC += else
EXEC +=    exit 0;

