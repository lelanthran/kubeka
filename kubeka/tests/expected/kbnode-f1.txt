Node [cron] with parent [(nil)]: 0x0
   appname: [ $(env_read APPNAME) ]
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   git_branch: [ master ]
   _LINE: [ 1 ]
   git_origin: [ ssh://some-origin/some-repo.git ]
Node [job] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   for_filter: [ first node ]
   JOBS: [ build ,upgrade ,Done ]
   _LINE: [ 7 ]
   MESSAGE: [ Performing a full deployment ]
   ID: [ Full deployment ]
Node [cron] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   for_filter: [ second node ]
   JOBS: [ Check changes ,checkout ,Build thing ,test ,package ]
   _LINE: [ 13 ]
   MESSAGE: [ Starting build process ]
   ID: [ build ]
Node [job] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   for_filter: [ third node ]
   JOBS: [ pre-deployment ,Upgrade DB ,Copy files ]
   _LINE: [ 19 ]
   MESSAGE: [ Performing upgrade ]
   ID: [ upgrade ]
Node [cron] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   _LINE: [ 25 ]
   EXEC: [  if [ `git  ls-remote $(git_origin) heads/master | cut -f 1 ` == $(current HASH)]; then exit -1; else exit 0; ]
   MESSAGE: [ Checking if sources changed, will continue only if sources changed ]
   ID: [ Check changes ]
f1 ====================================
Node [cron] with parent [(nil)]: 0x0
   appname: [ $(env_read APPNAME) ]
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   git_branch: [ master ]
   _LINE: [ 1 ]
   git_origin: [ ssh://some-origin/some-repo.git ]
Node [cron] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   for_filter: [ second node ]
   JOBS: [ Check changes ,checkout ,Build thing ,test ,package ]
   _LINE: [ 13 ]
   MESSAGE: [ Starting build process ]
   ID: [ build ]
Node [cron] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   _LINE: [ 25 ]
   EXEC: [  if [ `git  ls-remote $(git_origin) heads/master | cut -f 1 ` == $(current HASH)]; then exit -1; else exit 0; ]
   MESSAGE: [ Checking if sources changed, will continue only if sources changed ]
   ID: [ Check changes ]
f2 ====================================
Node [job] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   for_filter: [ first node ]
   JOBS: [ build ,upgrade ,Done ]
   _LINE: [ 7 ]
   MESSAGE: [ Performing a full deployment ]
   ID: [ Full deployment ]
Node [cron] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   for_filter: [ second node ]
   JOBS: [ Check changes ,checkout ,Build thing ,test ,package ]
   _LINE: [ 13 ]
   MESSAGE: [ Starting build process ]
   ID: [ build ]
Node [job] with parent [(nil)]: 0x0
   _FILENAME: [ ./tests/input/kbnode-f1.txt ]
   for_filter: [ third node ]
   JOBS: [ pre-deployment ,Upgrade DB ,Copy files ]
   _LINE: [ 19 ]
   MESSAGE: [ Performing upgrade ]
   ID: [ upgrade ]
